circup/__init__.py

```diff
@@ -1396,22 +1876,29 @@ def install(ctx, modules, pyext, requirement, auto, auto_file):  # pragma: no co
         # pass a local file with "./" or "../"
         is_relative = auto_file.split(os.sep)[0] in [os.path.curdir, os.path.pardir]
         if not os.path.isabs(auto_file) and not is_relative:
-            auto_file = os.path.join(ctx.obj["DEVICE_PATH"], auto_file or "code.py")
-        if not os.path.isfile(auto_file):
+            if not ctx.obj["using_webworkflow"]:
+                auto_file = os.path.join(ctx.obj["DEVICE_PATH"], auto_file or "code.py")
+            else:
+                auto_file = os.path.join(
+                    ctx.obj["DEVICE_PATH"], "fs", auto_file or "code.py"
+                )
+        if not os.path.isfile(auto_file) and not ctx.obj["using_webworkflow"]:
             click.secho(f"Auto file not found: {auto_file}", fg="red")
             sys.exit(1)
-        requested_installs = libraries_from_imports(auto_file, mod_names)
+        requested_installs = ctx.obj["backend"].libraries_from_imports(
```

Comment:
This seems like something that should be done outside of a backend. It should only depend on the contents of the file to read for auto detection.